src/openrct2-ui/windows/GameBottomToolbar.cpp

```diff
@@ -69,687 +69,672 @@ static Widget window_game_bottom_toolbar_widgets[] =
+            case WIDX_MONEY:
+                if (!(gParkFlags & PARK_FLAGS_NO_MONEY))
+                    ContextOpenWindow(WindowClass::Finances);
+                break;
+            case WIDX_GUESTS:
+                ContextOpenWindowView(WV_PARK_GUESTS);
+                break;
+            case WIDX_PARK_RATING:
+                ContextOpenWindowView(WV_PARK_RATING);
+                break;
+            case WIDX_MIDDLE_INSET:
+                if (News::IsQueueEmpty())
+                {
+                    ContextOpenWindow(WindowClass::RecentNews);
+                }
+                else
+                {
+                    News::CloseCurrentItem();
+                }
+                break;
+            case WIDX_NEWS_SUBJECT:
+                newsItem = News::GetItem(0);
+                News::OpenSubject(newsItem->Type, newsItem->Assoc);
+                break;
+            case WIDX_NEWS_LOCATE:
+                if (News::IsQueueEmpty())
+                    break;
 
-    // Draw money
-    if (!(gParkFlags & PARK_FLAGS_NO_MONEY))
-    {
-        Widget widget = window_game_bottom_toolbar_widgets[WIDX_MONEY];
-        auto screenCoords = ScreenCoordsXY{ w->windowPos.x + widget.midX(),
-                                            w->windowPos.y + widget.midY() - (line_height == 10 ? 5 : 6) };
+                {
+                    newsItem = News::GetItem(0);
 
-        colour_t colour
-            = (gHoverWidget.window_classification == WindowClass::BottomToolbar && gHoverWidget.widget_index == WIDX_MONEY
-                   ? COLOUR_WHITE
-                   : NOT_TRANSLUCENT(w->colours[0]));
-        StringId stringId = gCash < 0 ? STR_BOTTOM_TOOLBAR_CASH_NEGATIVE : STR_BOTTOM_TOOLBAR_CASH;
-        auto ft = Formatter();
-        ft.Add<money64>(gCash);
-        DrawTextBasic(*dpi, screenCoords, stringId, ft, { colour, TextAlignment::CENTRE });
-    }
+                    auto subjectLoc = News::GetSubjectLocation(newsItem->Type, newsItem->Assoc);
 
-    static constexpr const StringId _guestCountFormats[] = {
-        STR_BOTTOM_TOOLBAR_NUM_GUESTS_STABLE,
-        STR_BOTTOM_TOOLBAR_NUM_GUESTS_DECREASE,
-        STR_BOTTOM_TOOLBAR_NUM_GUESTS_INCREASE,
-    };
+                    if (!subjectLoc.has_value())
+                        break;
 
-    static constexpr const StringId _guestCountFormatsSingular[] = {
-        STR_BOTTOM_TOOLBAR_NUM_GUESTS_STABLE_SINGULAR,
-        STR_BOTTOM_TOOLBAR_NUM_GUESTS_DECREASE_SINGULAR,
-        STR_BOTTOM_TOOLBAR_NUM_GUESTS_INCREASE_SINGULAR,
-    };
+                    WindowBase* mainWindow = WindowGetMain();
+                    if (mainWindow != nullptr)
+                        WindowScrollToLocation(*mainWindow, subjectLoc.value());
+                }
+                break;
+            case WIDX_RIGHT_OUTSET:
+            case WIDX_DATE:
+                ContextOpenWindow(WindowClass::RecentNews);
+                break;
+            default:
+            break;
```

Comment:
There was no `default` before, any reason for adding one now?

Reply:
Not sure, think it might've been added automatically when I was moving around the code. I can remove it, if that would be better.